pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -379,27 +379,6 @@ func (configgen *ConfigGeneratorImpl) createGatewayHTTPFilterChainOpts(node *mod
 	routeName string, proxyConfig *meshconfig.ProxyConfig) *filterChainOpts {
 	serverProto := protocol.Parse(port.Protocol)
 
-	httpProtoOpts := &core.Http1ProtocolOptions{}
-
-	if features.HTTP10 || node.Metadata.HTTP10 == "1" {
-		httpProtoOpts.AcceptHttp_10 = true
-	}
-
-	xffNumTrustedHops := uint32(0)
-	forwardClientCertDetails := util.MeshConfigToEnvoyForwardClientCertDetails(meshconfig.Topology_SANITIZE_SET)
-
-	if proxyConfig != nil && proxyConfig.GatewayTopology != nil {
-		xffNumTrustedHops = proxyConfig.GatewayTopology.NumTrustedProxies
-		if proxyConfig.GatewayTopology.ForwardClientCertDetails != meshconfig.Topology_UNDEFINED {
-			forwardClientCertDetails = util.MeshConfigToEnvoyForwardClientCertDetails(proxyConfig.GatewayTopology.ForwardClientCertDetails)
-		}
-	}
-
-	var stripPortMode *hcm.HttpConnectionManager_StripAnyHostPort = nil
-	if features.StripHostPort {
-		stripPortMode = &hcm.HttpConnectionManager_StripAnyHostPort{StripAnyHostPort: true}
-	}
-
```

Comment:
Would suggest not weaving these unrelated changes

Reply:
This was added because we are building http connection manager twice and I had to set that option at both places. So I have refactored it to single function so that I can set once. If you think it is better to do it in a separate PR - I am fine, but it is some what related to this change.